ister_gui.py

```diff
@@ -2880,33 +2930,48 @@ def __init__(self, *args, **kwargs):
     def _init_actions(self):
         # pylint: disable=too-many-locals
         # pylint: disable=too-many-statements
+        current = itertools.count(start=1)
         # auto install assumed
+        total = 6
         keyboard_select = KeyboardSelection()
-        network_requirements = NetworkRequirements(1, 6,
+        network_requirements = NetworkRequirements(next(current), total,
                                                    self.args['contenturl'],
                                                    self.args['versionurl'])
-        choose_action = ChooseAction(2, 6)
-        telem_disclosure = TelemetryDisclosure(3, 6)
-        startmenu = StartInstaller(4, 6)
-        automatic_device = SelectDeviceStep(5, 6)
-        confirm_disk_wipe = ConfirmDiskWipe(6, 6)
+        choose_action = ChooseAction(next(current), total)
+        telem_disclosure = TelemetryDisclosure(next(current), total)
+        startmenu = StartInstaller(next(current), total)
+        automatic_device = SelectDeviceStep(next(current), total)
+        confirm_disk_wipe = ConfirmDiskWipe(next(current), total)
         # manual install
-        part_menu = PartitioningMenu(5, 12)
-        manual_part_device = SelectMultiDeviceStep(6, 12)
-        manual_nopart_device = SelectDeviceStep(6, 12)
+        current = itertools.count(start=5)
+        total = 13
+        part_menu = PartitioningMenu(next(current), total)
+        # Partition meta-step
+        step = next(current)
+        manual_part_device = SelectMultiDeviceStep(step, total)
+        manual_nopart_device = SelectDeviceStep(step, total)
         terminal_cgdisk = TerminalStep()
-        set_mount_points = MountPointsStep(7, 12)
-        confirm_disk_wipe2 = ConfirmDiskWipe(7, 12)
-        config_hostname = ConfigureHostname(8, 12)
-        confirm_user = ConfirmUserMenu(9, 12)
-        user_configuration = UserConfigurationStep(9, 12)
-        bundle_selector = BundleSelectorStep(10, 12)
-        confirm_dhcp = ConfirmDHCPMenu(11, 12)
-        static_ip_config = StaticIpStep(11, 12)
+        # Mount points meta-step
+        step = next(current)
+        set_mount_points = MountPointsStep(step, total)
+        confirm_disk_wipe2 = ConfirmDiskWipe(step, total)
+        config_cmdline = ConfigureCmdline(next(current), total)
+        config_hostname = ConfigureHostname(next(current), total)
+        # User configuration meta-step
+        step = next(current)
+        confirm_user = ConfirmUserMenu(step, total)
+        user_configuration = UserConfigurationStep(step, total)
+        bundle_selector = BundleSelectorStep(next(current), total)
+        # Networking meta-step
+        step = next(current)
+        confirm_dhcp = ConfirmDHCPMenu(step, total)
+        static_ip_config = StaticIpStep(step, total)
```

Comment:
I like the use of the counter but it gets a little confusing to me when the code alternates between next(current) and step when setting up the orderings. Maybe some reordering or more descriptive variable names as I see this being a possible issue for somebody who wants to change the list order in the future figuring out where to insert what potentially.

Reply:
I have to save next(current) to a variable if I want to use the int more than once before it is incremented.

I'll change the variable names and make the comments more verbose.

Reply:
Thanks